frontend/react/style-guide.md

```diff
@@ -159,38 +176,38 @@ src
 ## Naming
 
   - **Extensions**: Use `.js` extension for React components.
-  - **Filename**: For component filenames and services use PascalCase. E.g., `ReservationCard.js`.
+  - **Filename**: For services use PascalCase. E.g., `ReservationCard.js` or a folder with the service name and `index.js` as filename. For React components, there must be a folder in PascalCase with its name and the component file should be `index.js` (and optionally `layout.js` if you're using [Smart/Dumb components](https://medium.com/@thejasonfile/dumb-components-and-smart-components-e7b33a698d43)
   - **Reference Naming**: Use PascalCase for React components and camelCase for their associated elements. eslint: [`react/jsx-pascal-case`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-pascal-case.md)
 
     ```jsx
     // bad
-    import reservationCard from './ReservationCard';
+    import myService from './MyService';
+    import myComponent from './MyComponent';
 
     // good
-    import ReservationCard from './ReservationCard';
+    import MyService from './MyService'; # webpack infers index.js
+    import MyComponent from './MyComponent'; # webpack infers index.js
 
     // bad
-    const ReservationItem = <ReservationCard />;
+    const MyComponent = <MyComponent />;
 
     // good
-    const reservationItem = <ReservationCard />;
+    const myComponent = <MyComponent />;
     ```
   - **Component Hierarchy**: 
     - Component files should be inside folders that match the component's name.
     - Use index.js as the filename of a container component. Use `Container` as the suffix of the component's name.
-    - Use layout.js as the filename of a layout component.
+    - Use layout.js as the filename of a layout component. Only do this if your component is becoming too big, it should be an exception, not a rule.
```

Comment:
I's change this `... is becoming too big ...` for `... is clearly separable into smart and dumb components ...`

Reply:
In my point of view, the `layout.js` actually in this era of hooks that i hope we use in wolox is becoming a little bit useless, i would use smart & dumb pattern only in two cases:

1. The component layout is becoming too big and is a class component with many logic inside.
2. You are using a effect or a `componentDidMount` api call and you need to wrap the component with a loader HOC, in this case is mandatory to use `layout.js`

Reply:
I think I agree with alejo in this. Smart/Dumb, hooks or not, is being abused when it should be an exception, not a rule. It's bad to abuse it because it almost duplicates the files and it also increases the code size. 
I'm not sure if `use layout unless is clearly separable into smart and dumb components` is a good criteria because it's too subjective. We now have a linter rule that defined max lines. If you break that rule, you ma use this. I think that's the most objective reason to use it, other that the HOC rule alejo mentioned.

I'm inclined to keep the styleguide as short as possible, that's why `is becoming too big` seemed the best way to put it, but if you think it's better to go in depth in the explanation or change the wording I'm open to other suggestions.

Reply:
Having a little more experience in hooks i think that the smart/dumb component will be deprecated when using them with apollo/context. When using this two combined the components will tend to have their own logic and to be small organisms with their small behavior. Until them i agree with capi in maintaining the rule to be an exception and not a common practice. I will add to this file that it is a `must` to divide when the eslint error: max-lines appears

Reply:
I agree with Lucas on this. I think that if a component is getting too big, the solution is to divide it into smaller components, not divide it into index and layout. Probably the components further down the line will be more about layout and the ones at the top will have more state managing-related stuff.
I don't think there's any reason to keep layout.js files around. In any case, they would be index.js files with no logic.